app/models/bookings/school_search.rb

```diff
@@ -127,6 +127,17 @@ def radius=(dist)
   def save_with_result_count(count)
     self.number_of_results = count
     save
+  rescue ActiveRecord::StatementInvalid => e
+    # This is a temporary measure to pin down a PG::CharacterNotInRepertoire exception
+    # that is occasionally raised on saving this model. I'm not sure how it gets into
+    # this state yet, but I expect its someone/a bot with JS disabled putting unexpected
+    # characters into the input field.
```

Comment:
Should the input fields not be sanitised before being passed into an SQL statement ? I expect the UTF-8 encoding means that unicode characters will get through, but then be rejected by Postgres if the encoding on Postgres is ACSII only.

Reply:
It's a bit of a mystery to me this one; what I _think_ is happening is ASCII characters are somehow getting into the `query` attribute of this model. How that's happening I'm not sure (hopefully the log line here will help); Rails is configured for UTF8 as is our Postgres DB and ActiveRecord is going to be sanitising SQL statements (but not for ASCII characters, I guess)